projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts

```diff
@@ -152,6 +156,39 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
         this.scrollToBottom();
     }
 
+    private handleQueryParams(): void {
+        this.route.queryParams
+            .pipe(
+                filter((params) => params.id),
+                zipWith(this.userService.getSessionTokenForRequest()),
+                mergeMap(([params, sessionToken]) =>
+                    this.http.get<EbvDataset>(`${this.config.API_URL}/ebv/dataset/${params.id}`, {
+                        headers: new HttpHeaders().set('Authorization', `Bearer ${sessionToken}`),
+                    }),
+                ),
+            )
+            .subscribe((dataset) => {
+                const ebvClass = this.ebvClasses.find((c) => c.name === dataset.ebvClass);
+
+                if (!!ebvClass) {
```

Comment:
You could use some early returns in order to remove two of the four layers of hierarchy in this method.

Despite that, should we show some notification when this went wrong or do we silently ignore it?